probe/probe.go

```diff
@@ -0,0 +1,315 @@
+            return err
+        }
+	}else{
+        checkHostAuditSupport()
+        checkLsmSupport(getHostSupportedLSM())
+        color.Blue("To get full probe, a daemonset will be deployed in your cluster - This daemonset will be deleted after probing")
+        color.Blue("Use --full tag to get full probing")
+    }
+    return nil
+}
+
+
+func checkLsmSupport(supportedLSM string) {
+    fmt.Printf("\t Enforcement:") 
+    if strings.Contains(supportedLSM, "selinux"){
+        color.Yellow(" Partial (Supported LSMs: "+ supportedLSM + ") \n\t To have full enforcement support, apparmor must be supported")
+    }else if strings.Contains(supportedLSM, "apparmor"){
+        color.Green(" Full (Supported LSMs: "+ supportedLSM + ")")
+    }else{
+        color.Red(" None (Supported LSMs: "+ supportedLSM + ") \n\t To have full enforcement support, apparmor must be supported")
+    }
+}
+
+func getHostSupportedLSM() string {
+
+    b, err := os.ReadFile("/sys/kernel/security/lsm")
+    if err != nil {
+        log.Printf("an error occured when reading file")
+        return "none"
+    }
+    s := string(b)
+    return s
+}
+
+
+
+func kernelVersionSupported(kernelVersion string) bool {
+    if semver.Compare(kernelVersion, "4.14") >= 0 {
+        return true
+    }
+    return false;
+}
+
+func checkAuditSupport(kernelVersion string, kernelHeaderPresent bool) {
+
+    if kernelVersionSupported(kernelVersion) && kernelHeaderPresent{
+        color.Green(" Supported (Kernel Version " + kernelVersion )
+    }else if(kernelVersionSupported(kernelVersion)){
+        color.Red(" Not Supported : Kernel header must be present")
+    }else{
+        color.Red(" Not Supported (Kernel Version " + kernelVersion + " \n\t Kernel version must be greater than 4.14) and kernel header must be present")
+    }
+
+}
+
+func checkKernelHeaderPresent() bool{
+    //check if there's any directory /usr/src/$(uname -r)
+    var uname syscall.Utsname
+    if err := syscall.Uname(&uname); err == nil {
+
+        var path = ""
+        if _, err := os.Stat("/etc/redhat-release"); !os.IsNotExist(err) {
+            path = "/usr/src/"+int8ToStr(uname.Release[:])
+        }else{
+            path = "/usr/src/linux-headers-"+int8ToStr(uname.Release[:])
+        }
+       
+        if _, err := os.Stat(path); !os.IsNotExist(err) {
+            return true
+        }        
+        
+	}
+     return false
+}
+
+/** check if there's any file like $(uname -r) in directory /usr/src/  **/
+func checkNodeKernelHeaderPresent(c *k8s.Client, nodeName string, kernelVersion string) bool{
+    pods, err := c.K8sClientset.CoreV1().Pods("default").List(context.Background(), metav1.ListOptions{
+		LabelSelector: "k8s-app=karmor-probe",
+        FieldSelector: "spec.nodeName=" + nodeName,
+	})
+	if err != nil {
+		return false
+	}
+    srcPath :=  "/usr/src/"
+    fileName := "*"+kernelVersion+"*"
+    reader, outStream := io.Pipe()
+
+    cmdArr := []string{"find", srcPath, "-name", fileName}
```

Comment:
Wrt above review comment, will need to handle lib modules check here as well ig.

Reply:
> Wrt above review comment, will need to handle lib modules check here as well ig.

pending